src/ophyd_async/epics/areadetector/writers/general_hdffile.py

```diff
@@ -17,8 +17,9 @@
 class _HDFDataset:
     data_key: str
     dataset: str
-    swmr: bool = False
+    shape: Sequence[int] = field(default_factory=tuple)
     multiplier: int = 1
+    swmr: bool = False
 
```

Comment:
Correction to the commit message - add back shape and move `swmr` to the `bottom`